### PR TITLE
feat: upgrade jQuery to v3

### PR DESCRIPTION
**Pull request by fibble**
## Description

Update Jquery to v3

Related issue: [VOL-5555](https://dvsa.atlassian.net/browse/VOL-5555)

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [x] Have you have added tests that prove the fix or feature is effective and working
- [x] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "grunt-contrib-uglify": "^5.2.2",
     "grunt-sass": "^3.1.0",
     "grunt-svg-sprite": "^2.0.2",
-    "jquery": "1.12.4",
+    "jquery": "^3.7.1",
```
Comment (JoshuaLicense) [inline package.json:25]: Are we missing a commit for the lockfile? NPM library versions are determined from the lockfile and the PR doesn't have a lockfile update.

File: assets/_js/init/selfserve.js [hunk 2]
```diff
@@ -14,8 +14,6 @@ OLCS.ready(function() {
 
   OLCS.accessibility();
 
-  OLCS.searchFilter();
-
```
Comment (JoshuaLicense) [inline assets/_js/init/selfserve.js:18]: How come we removed this?

File: assets/_js/components/browser.js [hunk 3]
```diff
@@ -13,15 +13,17 @@ OLCS.browser = (function(document, $, undefined) {
...
-  exports.isSafari = Object.prototype.toString.call(window.HTMLElement).indexOf("Constructor") > 0;
+  exports.isSafari = (/constructor/i.test(window.HTMLElement) ||
+      (function (p) { return p.toString() === "[object SafariRemoteNotification]"; })
+      (typeof window.safari !== "undefined" && typeof window.safari.pushNotification !== "undefined"));
 
-  exports.isChrome = !!window.chrome && !exports.isOpera;
+  exports.isChrome = (!!window.chrome && !exports.isOpera && /Google Inc/.test(navigator.vendor));
 
-  exports.isIE = /*@cc_on!@*/false || !!document.documentMode;
+  exports.isIE = (/*@cc_on!@*/false || !!document.documentMode);
```
Comment (JoshuaLicense) [inline assets/_js/components/browser.js:26]: Was this broken? This isn't related to jQuery is it?

File: Gruntfile.js [hunk 4]
```diff
@@ -39,8 +39,8 @@ var sass = require('sass');
         // Function to get all scripts for use with a given theme
         var scriptPaths = function(theme) {
             var files = [
-                'node_modules/jquery/dist/jquery.min.js',
-                'node_modules/chosen-npm/public/chosen.jquery.min.js',
+                'node_modules/jquery/dist/jquery.js',
+                'node_modules/chosen-npm/public/chosen.jquery.js',
```
Comment (JoshuaLicense) [inline Gruntfile.js:43]: We probably should use the minimised version of jQuery.

**Review by JoshuaLicense — CHANGES_REQUESTED** [review]

File: Gruntfile.js (hunk 4 above)
Comment (fibble) [reply Gruntfile.js:43]: Agreed, switched to this to make readings some console errors easier. Reverting to min.

File: assets/_js/components/browser.js (hunk 3 above)
Comment (fibble) [reply assets/_js/components/browser.js:26]: Deprecation warnings were present in the console regarding some of the checks.

**Review by JoshuaLicense — APPROVED** [review]
I expected a fair bit of changes, but pleasantly surprised. Nice work!

**Review by JoshuaLicense — APPROVED** [review]
I was expecting a few more changes to the JavaScript to be honest, but pleasantly surprised.

**Review by gabrielg2020 — APPROVED** [review]

**Pull request merged**